packages/server/lib/util/print-run.ts

```diff
@@ -0,0 +1,536 @@
+  ]).join(' ')
+}
+
+function formatFooterSummary (results) {
+  const { totalFailed, runs } = results
+
+  const isCanceled = _.some(results.runs, { skippedSpec: true })
+
+  // pass or fail color
+  const c = isCanceled ? 'magenta' : totalFailed ? 'red' : 'green'
+
+  const phrase = (() => {
+    if (isCanceled) {
+      return 'The run was canceled'
+    }
+
+    // if we have any specs failing...
+    if (!totalFailed) {
+      return 'All specs passed!'
+    }
+
+    // number of specs
+    const total = runs.length
+    const failingRuns = _.filter(runs, 'stats.failures').length
+    const percent = Math.round((failingRuns / total) * 100)
+
+    return `${failingRuns} of ${total} failed (${percent}%)`
+  })()
+
+  return [
+    isCanceled ? '-' : formatSymbolSummary(totalFailed),
+    color(phrase, c),
+    gray(duration.format(results.totalDuration)),
+    colorIf(results.totalTests, 'reset'),
+    colorIf(results.totalPassed, 'green'),
+    colorIf(totalFailed, 'red'),
+    colorIf(results.totalPending, 'cyan'),
+    colorIf(results.totalSkipped, 'blue'),
+  ]
+}
+
+function formatSymbolSummary (failures) {
+  return failures ? logSymbols.error : logSymbols.success
+}
+
+function macOSRemovePrivate (str) {
+  // consistent snapshots when running system tests on macOS
+  if (process.platform === 'darwin' && str.startsWith('/private')) {
+    return str.slice(8)
+  }
+
+  return str
+}
+
+function collectTestResults (obj: { video?: boolean, screenshots?: Screenshot[] }, estimated) {
+  return {
+    name: _.get(obj, 'spec.name'),
+    baseName: _.get(obj, 'spec.baseName'),
+    tests: _.get(obj, 'stats.tests'),
+    passes: _.get(obj, 'stats.passes'),
+    pending: _.get(obj, 'stats.pending'),
+    failures: _.get(obj, 'stats.failures'),
+    skipped: _.get(obj, 'stats.skipped'),
+    duration: humanTime.long(_.get(obj, 'stats.wallClockDuration')),
+    estimated: estimated && humanTime.long(estimated),
+    screenshots: obj.screenshots && obj.screenshots.length,
+    video: Boolean(obj.video),
+  }
+}
+
+function formatPath (name, n, colour = 'reset') {
+  if (!name) return ''
+
+  const fakeCwdPath = env.get('FAKE_CWD_PATH')
+
+  if (fakeCwdPath && env.get('CYPRESS_INTERNAL_ENV') === 'test') {
+    // if we're testing within Cypress, we want to strip out
+    // the current working directory before calculating the stdout tables
+    // this will keep our snapshots consistent everytime we run
+    const cwdPath = process.cwd()
+
+    name = name
+    .split(cwdPath)
+    .join(fakeCwdPath)
+
+    name = macOSRemovePrivate(name)
+  }
+
+  // add newLines at each n char and colorize the path
+  if (n) {
+    let nameWithNewLines = newlines.addNewlineAtEveryNChar(name, n)
+
+    return `${color(nameWithNewLines, colour)}`
+  }
+
+  return `${color(name, colour)}`
+}
+
+function formatNodeVersion ({ resolvedNodeVersion, resolvedNodePath }: Pick<Cfg, 'resolvedNodeVersion' | 'resolvedNodePath'>, width) {
```

Comment:
Can we split out / add the appropriate unit tests for this? The snapshots exist but are more a side-effect of other aspects vs explicitly testing these.

Also is it possible to fully type this new file verse adding partial types?

Reply:
The goal of this PR is to just split out some LoC from `run.ts` to make it easier to work in.

All of the `modes/run.js` unit tests were removed during unification, so there is not currently any unit-level testing for this logic. There's a ton of logic in here that would be helpful to have unit tests for, but since we already have coverage for this in the form of system-test snapshots, I don't feel it's particularly important to add in this PR, since it has survived without unit tests for this long.

I added more types to make this area easier to work in: e0aeb06ef2 I started trying to type the `results` object, but it turned into a rabbithole, it's overloaded with 3 separate result types (Mocha end, early exit, and dashboard skip), properly typing it would require either a refactor or a lot of falsy checks.